### PR TITLE
KAA-1283: [C++ SDK] Review all C++ code samples in new Documentation

### PR DESCRIPTION
**Pull request by sparadiuk**

### CODE REVIEW

File: doc/Programming-guide/Your-first-Kaa-application/index.md [hunk 1]
```diff
@@ -381,13 +409,110 @@ int main(void)
...
     return EXIT_SUCCESS;
 }
-
 ```
 
 </div>
+ 
+<div id="app-cpp" class="tab-pane fade" markdown="1" >
+```c++
+class ConfigurationCollection : public kaa::IConfigurationReceiver {
```
Comment (Dubland) [inline doc/Programming-guide/Your-first-Kaa-application/index.md:478]: Add `#include` for header files

File: doc/Programming-guide/Your-first-Kaa-application/index.md [hunk 2]
```diff
@@ -182,27 +182,56 @@ Before you start with C application code, some preparation is required.
...
+
+Before you start with C++ application code, some preparation is required:
+
+1. Need to install the next libraries:  [CMake, Boost, AvroC++, Botan]({{root_url}}/Programming-guide/Using-Kaa-endpoint-SDKs/C++/SDK-Linux/#installing-prerequisites)
+
+2. Create `kaa` directory and unpack C++ SDK
+3. Create `CMakeLists.txt` in the application directory with following contents:
+
+        cmake_minimum_required(VERSION 2.8.12)
+        project (Cpp-SDK-your-first-Kaa-application CXX)
```
Comment (Dubland) [inline doc/Programming-guide/Your-first-Kaa-application/index.md]: Remove space before parenthesis.

File: doc/Programming-guide/Your-first-Kaa-application/index.md [hunk 3]
```diff
@@ -404,30 +529,41 @@ int main(void)
...
         cd build
         cmake -DKAA_MAX_LOG_LEVEL=3 ..
         make
 
-2. Launch the executable file.
+2. Launch the executable file:
 
         ./kaa-app
-
+
```
Comment (Dubland) [inline doc/Programming-guide/Your-first-Kaa-application/index.md]: Remove trailing whitespace

File: doc/Programming-guide/Key-platform-features/Notifications/index.md [hunk 4]
```diff
@@ -754,22 +753,19 @@ kaaClient.removeNotificationListener("All devices notifications", listener);
...
-#include <iostream>
-#include <string>
 #include <memory>
- 
 #include <kaa/Kaa.hpp>
  
 ...
  
-// Create the listener which receives notifications on the specified optional topic.
+// Create the listener which receives notifications on the specified optional topic
```
Comment (Dubland) [inline doc/Programming-guide/Key-platform-features/Notifications/index.md]: Why you remove this dot?

File: doc/Programming-guide/Key-platform-features/Endpoint-ownership/index.md [hunk 5]
```diff
@@ -504,39 +495,35 @@ kaaClient.detachEndpoint("endpointKeyHash", new OnDetachEndpointOperationCallbac
...
 // Create an endpoint instance
-auto kaaClient = Kaa::newClient();
+auto kaaClient = kaa::Kaa::newClient();
  
-// Start an endpoint
-kaaClient->start();
+...
  
-// Try to attach an endpoint to a user
+// Try to detach an endpoint
```
Comment (Dubland) [inline doc/Programming-guide/Key-platform-features/Endpoint-ownership/index.md]: Trailing whitespace

File: doc/Programming-guide/Your-first-Kaa-application/index.md [hunk 6]
```diff
@@ -165,15 +165,15 @@ The following code block illustrates a simple desktop application that sends vir
...
 <div id="prep-c" class="tab-pane fade in active" markdown="1" >
 
-Before you start with C application code, some preparation is required.
+Before you start with C application code, some preparation is required:
 
-1. Install dependencies: CMake. On Ubuntu, you can install it using following commands.
+1. Install dependencies: CMake. On Ubuntu, you can install it using following command:
 
         sudo apt-get install cmake
 1. Create `kaa` directory and unpack C SDK
```
Comment (Dubland) [inline doc/Programming-guide/Your-first-Kaa-application/index.md]: Change '1' to '2'

File: doc/Programming-guide/Your-first-Kaa-application/index.md [hunk 7]
```diff
@@ -165,15 +165,15 @@ The following code block illustrates a simple desktop application that sends vir
...
-1. Create `CMakeLists.txt` in the application directory with following contents.
+1. Create `CMakeLists.txt` in the application directory with following contents:
```
Comment (Dubland) [inline doc/Programming-guide/Your-first-Kaa-application/index.md]: Change '1' to '3'

File: doc/Programming-guide/Your-first-Kaa-application/index.md [hunk 8]
```diff
@@ -182,27 +182,56 @@ Before you start with C application code, some preparation is required.
 
         add_executable(kaa-app main.c)
         target_link_libraries(kaa-app kaac)
-1. Create source file `main.c` with empty main routine (for now).
+1. Create source file `main.c` with empty main routine (for now):
```
Comment (Dubland) [inline doc/Programming-guide/Your-first-Kaa-application/index.md]: Change '1' to '4'

File: doc/Programming-guide/Your-first-Kaa-application/index.md (hunk 6 above)
Comment (RostakaGmfun) [reply doc/Programming-guide/Your-first-Kaa-application/index.md]: It's OK, Markdown will do it on its own

File: doc/Programming-guide/Your-first-Kaa-application/index.md (hunk 7 above)
Comment (RostakaGmfun) [reply doc/Programming-guide/Your-first-Kaa-application/index.md]: Same as above

File: doc/Programming-guide/Your-first-Kaa-application/index.md (hunk 8 above)
Comment (RostakaGmfun) [reply doc/Programming-guide/Your-first-Kaa-application/index.md]: Same

Comment (Dubland) [reply doc/Programming-guide/Your-first-Kaa-application/index.md]: But in the next lines '5' and '6'

**Comment by vadimol** [comment]
Please, rebase it on the latest develop branch

File: doc/Programming-guide/Your-first-Kaa-application/index.md (hunk 1 above)
Comment (sparadiuk) [reply doc/Programming-guide/Your-first-Kaa-application/index.md:478]: added

File: doc/Programming-guide/Your-first-Kaa-application/index.md (hunk 2 above)
Comment (sparadiuk) [reply doc/Programming-guide/Your-first-Kaa-application/index.md]: Changed

File: doc/Programming-guide/Your-first-Kaa-application/index.md (hunk 3 above)
Comment (sparadiuk) [reply doc/Programming-guide/Your-first-Kaa-application/index.md]: Removed

File: doc/Programming-guide/Your-first-Kaa-application/index.md (hunk 6 above)
Comment (sparadiuk) [reply doc/Programming-guide/Your-first-Kaa-application/index.md]: I don't mind to fix it, but we have another task for this.
http://jira.kaaproject.org/browse/KAA-1282?jql=text%20~%20%22Review%20all%20C%20code%20samples%20in%20new%20Documentation%22

File: doc/Programming-guide/Your-first-Kaa-application/index.md (hunk 7 above)
Comment (sparadiuk) [reply doc/Programming-guide/Your-first-Kaa-application/index.md]: The same as above

File: doc/Programming-guide/Your-first-Kaa-application/index.md (hunk 8 above)
Comment (sparadiuk) [reply doc/Programming-guide/Your-first-Kaa-application/index.md]: The same as above

File: doc/Programming-guide/Key-platform-features/Notifications/index.md (hunk 4 above)
Comment (sparadiuk) [reply doc/Programming-guide/Key-platform-features/Notifications/index.md]: returned

File: doc/Programming-guide/Key-platform-features/Endpoint-ownership/index.md (hunk 5 above)
Comment (sparadiuk) [reply doc/Programming-guide/Key-platform-features/Endpoint-ownership/index.md]: returned

**Comment by vadimol** [comment]
@kaa-leeroy retest this please

File: doc/Administration-guide/index.md [hunk 9]
```diff
@@ -22,3 +22,4 @@ view. Helps with planning your deployment, provides an overview of
 [Upgrading your instance]({{root_url}}Administration-guide/Upgrading-your-instance/) helps migrate to the new Kaa platform version.
 [Troubleshooting]({{root_url}}Administration-guide/Troubleshooting/) explains how to resolve some common issues while using Kaa as well as how to report issues
 to the Kaa Crew for getting help.
+
```
Comment (vadimol) [inline doc/Administration-guide/index.md:25]: This blank line is redundant.

**Review by vadimol — APPROVED** [review]

**Pull request merged**